pkgs/by-name/sw/swhkd/package.nix

```diff
@@ -0,0 +1,68 @@
+{ lib
+, rustPlatform
+, fetchurl
+, fetchFromGitHub
+, pkg-config
+, scdoc
+, libgcc
+, systemd
+}:
+rustPlatform.buildRustPackage rec {
+    pname = "swhkd";
+    version = "1.2.1-unstable-2024-04-06";
+
+    # split-output derivation, since there's a fair amount of associated data for
+    # pkexec and such.
+    outputs = [ "bin" "man" "out" ];
+
+    src = fetchFromGitHub {
+        owner = "waycrate";
+        repo = "swhkd";
+        # build from master, since the 1.2.1 makefile is unsutible for packaging
+        rev = "f8519a54900d72492a6c036b32e472c108d44dbf";
+        hash = "sha256-zyGyZOG8gAtsRkzSRH1M777fPv1wudbVsBrSTJ5CBnY=";
+      };
+
+    nativeBuildInputs = [
+      scdoc
+      pkg-config
+    ];
+
+    # the makefile tries to set the ownership of a file to root.
+    # this will fail, but files are owned by root anyways.
+    postPatch = "sed -ie 's/-o root//' Makefile";
+
+    buildPhase = ''
+      runHook preBuild
+      make build
+      runHook postBuild
+    '';
+
+    installPhase = ''
+      runHook preInstall
+
+      mkdir -p $out $bin $man/share
+      make DESTDIR=$out MAN1_DIR=/share/man/man1 MAN5_DIR=/share/man/man5 TARGET_DIR=/bin install
+      mv $out/bin $bin/bin
+      mv $out/share/man $man/share/man
+      mv $out/usr/share/polkit-1 $out/share/polkit-1
```

Comment:
To move things between outputs, please use moveToOutput. Also I think manpages should be moved automatically in fixupPhase

Reply:
moveToOutput is seemingly undocumented, and i think relying on implicit behavior would make the derivation harder to understand.

Reply:
That's the standard way to move files from different outputs and emulating that through other means is discouraged, especially if some functionality is missing https://github.com/NixOS/nixpkgs/blob/master/pkgs/build-support/setup-hooks/multiple-outputs.sh#L105-L149

There is maybe some documentation missing, but that doesn't allow disregarding functions from stdenv.